README.md

```diff
@@ -110,6 +110,46 @@ end
 
 ```
 
+#### Testing that a ruby method is called with the proper arguments
+
+This assumes you are testing ruby code, not puppet code.
+
+Code under test:
+```ruby
+Puppet::Functions.create_function(:eciton) do
+  dispatch :eciton do
+    param 'String', :some_string
+  end
+
+  def eciton(some_string)
+    Puppet.notice(some_string.reverse)
+  end
+end
+```
+
+To test that the correct string is being set to `notice`...
+
+In your `spec_helper`:
+```ruby
+RSpec.configure do |c|
+  c.mock_with :respec
```

Comment:
This could do with more emphasis. IMO `:rspec` (typo, btw!) should be the recommended default. These days the PDK supports this through `mock_with`: https://github.com/puppetlabs/pdk-templates/blob/046d65e6310c68c2209b7c41c73511b9821c1e79/moduleroot/spec/spec_helper.rb.erb#L31

See also https://github.com/puppetlabs/puppet-resource_api#getting-started for an example.